src/renderers/WebGLRenderer.js

```diff
@@ -1483,12 +1762,80 @@ function WebGLRenderer( parameters ) {
 
 			} else {
 
-				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group );
+				var instanceCount = undefined;
+
+				if ( geometry && geometry.isInstancedBufferGeometry && geometry.maxInstancedCount > 0 ) {
+
+					instanceCount = geometry.maxInstancedCount;
+
+				}
+
+				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group, instanceCount );
+
+			}
+
+			if ( object.onAfterRender ) {
+
+				object.onAfterRender( _this, scene, camera, geometry, material, group );
 
 			}
 
-			object.onAfterRender( _this, scene, camera, geometry, material, group );
+		}
+
+	}
+
+	function renderInstancedGeometryMap ( instancedGeometryMap, scene, camera, overrideMaterial ) {
+
+		for ( var uuid in instancedGeometryMap ) {
+
+			if ( instancedGeometryMap.hasOwnProperty( uuid ) ) {
+
+				var instancedGeometry = instancedGeometryMap[ uuid ];
+				var geometry = instancedGeometry.geometry;
+
+				for (var groupIndex = 0, groupCount = instancedGeometry.groups.length; groupIndex < groupCount; groupIndex++ ) {
+
+					var instancingGroup = instancedGeometry.groups[ groupIndex ];
+
+					var objects = instancingGroup.objects;
+					var material = overrideMaterial ? overrideMaterial : instancingGroup.material;
+
+					if ( objects.length > 0 ) {
+
+						var group = geometry.groups[ groupIndex ] || null;
 
+						for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+							var object = objects[ i ];
+
+							object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
+							object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
+
+						}
+
+						if ( objects.length < _this.minInstancingBatchSize ) {
+
+							for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+								_this.renderBufferDirect( camera, scene.fog, geometry, material, objects[ i ], group );
```

Comment:
I'm still familiarizing with the code but this line appears to be the source of a bug where some geometry faces start disappearing when going below the `minInstancingBatchSize` threshold. It appears that the `group` argument being passed may be of a different format than is expected by `this.renderBufferDirect` when going through this code path.